dianatanase/assignment2/Q1_tests.java

```diff
@@ -5,120 +5,94 @@
-		ArrayList<Integer> ancestors2and3 = new ArrayList<Integer>(Arrays.asList(1));
-		ArrayList<Integer> ancestors4and5 = new ArrayList<Integer>(Arrays.asList(2,1));
-		ArrayList<Integer> ancestors6and7 = new ArrayList<Integer>(Arrays.asList(3,1));
-		ArrayList<Integer> ancestors8and9 = new ArrayList<Integer>(Arrays.asList(4,2,1));
+		Node<Integer> root = new Node<Integer>(7,15);
+		Tree<Integer> tree = new Tree<Integer>();
+		tree.setRoot(root);
 		
-		// Test correct ancestors for each node.
-		assertEquals(ancestors1, Q1.getAncestorsInTree(1, tree));
-		assertEquals(ancestors2and3, Q1.getAncestorsInTree(2, tree));
-		assertEquals(ancestors2and3, Q1.getAncestorsInTree(3, tree));
-		assertEquals(ancestors4and5, Q1.getAncestorsInTree(4, tree));
-		assertEquals(ancestors4and5, Q1.getAncestorsInTree(5, tree));
-		assertEquals(ancestors6and7, Q1.getAncestorsInTree(6, tree));
-		assertEquals(ancestors6and7, Q1.getAncestorsInTree(7, tree));
-		assertEquals(ancestors8and9, Q1.getAncestorsInTree(8, tree));
-		assertEquals(ancestors8and9, Q1.getAncestorsInTree(9, tree));
+		try {
+			tree.addChildToKey(3,5,-1);
+			 fail("Adding child should fail if key of parent is not in tree");
+		} catch (Exception e) {
+		}
+	}
+	
+	// Test method fails for an empty or null tree.
+	@Test
+	public void testNull(){
 		
-		// Test with keys that are not accepted (negative) or do not exist in the tree.
-		assertEquals(ancestors1, Q1.getAncestorsInTree(-10, tree));
-		assertEquals(ancestors1, Q1.getAncestorsInTree(20, tree));
+		Tree<Integer> tree = new Tree<Integer>();
+		try {
+			 Q1.getAncestorsInTree(5, tree);
+			 fail("Method should fail with empty tree.");
+		} catch (Exception e) {
+		}
+	
+		try {
+			 Q1.getAncestorsInTree(5, null);
+			 fail("Method should fail with empty tree.");
+		} catch (Exception e) {
+		}		
 	}
 	
 	// Test with keys not in order and data of char type.
 	@Test
-	public void testKeysNotInOrder() {
+	public void testKeysInTree() throws Exception {
 		
-		// Initialize nodes.
-		Node<Character> node1 = new Node<Character>(null, 7, 'x', null, null);
-		Node<Character> node2 = new Node<Character>(node1, 3, 'a', null, null);
-		Node<Character> node3 = new Node<Character>(node1, 4, 'v', null, null);
-		Node<Character> node4 = new Node<Character>(node2, 2, 'w', null, null);
-		Node<Character> node5 = new Node<Character>(node2, 5, 't', null, null);
-		Node<Character> node6 = new Node<Character>(node3, 8, 'u', null, null);
-		Node<Character> node7 = new Node<Character>(node4, 1, 's', null, null);
-		Node<Character> node8 = new Node<Character>(node4, 6, 'p', null, null);
-
-		// Link nodes to their parents.s
-		node1.leftNode(node2);
-		node1.rightNode(node3);
+		Tree<Integer> tree = setUp();
+				
+		// Declare expected results and test correct ancestors for each node.
+		ArrayList<Integer> ancestors7 = new ArrayList<Integer>();
+		assertEquals(ancestors7, Q1.getAncestorsInTree(7, tree));
```

Comment:
A couple of pointers with asserts. It is standard practice to have the actualValue first, then the expectedValue second, and you can inline the Arrays.asList()
`assertEquals(Q1.getAncestorsInTree(3, tree), ArraysAsList(7))` should work.